### PR TITLE
docs: discourage memo usage

### PR DESCRIPTION
**Pull request by rootulp**
Add a CIP that discourages memo usage

### CODE REVIEW

**Review by jcstein — APPROVED** [review]

File: cips/discourage-memo-usage.md [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+> Application message retrieval partitioning. Client nodes must be able to download all of the messages relevant to the applications they use from storage nodes, without needing to downloading any messages for other applications.
+
+For this reason, blob data is scoped to an application-specific namespace. However, the transaction memo field is not scoped to an application-specific namespace. Instead, it pollutes the reserved [`TRANSACTION_NAMESPACE`](https://celestiaorg.github.io/celestia-app/specs/namespace.html#reserved-namespaces). This is undesirable because partial nodes that want to verify the current Celestia state must download and execute all transactions in the `TRANSACTION_NAMESPACE` (including memos).
+
+As of January 17, 2024, 0.51 GiB of data has been published to the Celestia network via memos and 1.46 GiB of data has been published via blob data.
+
+This proposal seeks to realign incentives so that protocol builders are encouraged to favor application-specific namespace blob data over memo fields.
+
```
Comment (jcstein) [inline cips/discourage-memo-usage.md:42]: - [ ] Add specification section with current values and values that will be changed to after this CIP

File: cips/discourage-memo-usage.md [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+Param                    | Celestia | Cosmos Hub | Osmosis
+-------------------------|----------|------------|--------
+`auth.MaxMemoCharacters` | 256      | 512        | 256
+`auth.TxSizeCostPerByte` | 10       | 10         | 10
+
+## Backwards Compatibility
+
+This proposal is backwards compatible. However, clients that hard-coded gas estimation based on the previous `auth.TxSizeCostPerByte` value will need to be updated.
+
```
Comment (jcstein) [inline cips/discourage-memo-usage.md:123]: - [ ] Add Test Cases section for testing this on Mocha testnet first

**Review by jcstein — CHANGES_REQUESTED** [review]

File: cips/discourage-memo-usage.md [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+## Security Considerations
```
Comment (jcstein) [inline cips/discourage-memo-usage.md:124]: - [ ] For reference implementation, maybe add the file that will be submitted to on-chain governance as a JSON file as this example:

```bash
{
  "title": "",
  "description": "",
  "deposit": [
    {
      "denom": "utia",
      "amount": ""
    }
  ]
}
```

File: cips/discourage-memo-usage.md [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+    Bithumb  | 10
+    Coinbase | 10
+    Gemini   | 13
+    KuCoin   | 10
+
+2. Some IBC relayers include the Hermes version in their memo. For [example](https://www.mintscan.io/celestia/tx/5FED84C1DA596EFC7F9005866573B31CC593770C6022B16B60834F1D22365E49?height=556792): `mzonder | hermes 1.7.4+ab73266 (https://hermes.informal.systems)` which is 64 characters.
+
+Given this context, what is the distribution of memo lengths in practice? How often are they used vs. empty?
+
+![Number of txs vs. memo length](../assets/memo-length.svg)
```
Comment (jcstein) [inline cips/discourage-memo-usage.md:63]:
```suggestion
![Number of txs vs. memo length](../assets/cip-14/memo-length.svg)
```

File: cips/discourage-memo-usage.md [hunk 5]
```diff
@@ -0,0 +1,139 @@
+---
```
Comment (jcstein) [inline cips/discourage-memo-usage.md:1]:
```suggestion
---
cip: 15
```

File: cips/discourage-memo-usage.md (hunk 1 above)
Comment (rootulp) [reply cips/discourage-memo-usage.md:43]: Good idea resolved via 99bdcb93558544fc205a0ff81e7d84ec3890416a

File: cips/discourage-memo-usage.md (hunk 2 above)
Comment (rootulp) [reply cips/discourage-memo-usage.md:131]: Resolved via 68d1080844120f285b5fe6a3d78eb3e4c9e83054

File: cips/discourage-memo-usage.md (hunk 3 above)
Comment (rootulp) [reply cips/discourage-memo-usage.md:181]: Thanks! Great idea. Added in a97aaefe304651e26c5ce2383a55f9a2926afec3

Based on this branch: https://github.com/celestiaorg/celestia-app/compare/main...rootulp:celestia-app:rp/discourage-memos-proposal

**Review by YazzyYaz — APPROVED** [review]

**Pull request merged**